src/dmap_import/util_api.py

```diff
@@ -0,0 +1,157 @@
+import os
+from typing import Optional, List, TypedDict
+import datetime
+import requests
+
+import sqlalchemy as sa
+
+from dmap_import.util_rds import DatabaseManager
+from dmap_import.schemas.api_metadata import ApiMetadata
+from dmap_import.util_logging import ProcessLogger
+
+
+class ApiResult(TypedDict):
+    """CUBIC API Result Model"""
+
+    id: str
+    dataset_id: str
+    url: str
+    start_date: str
+    end_date: str
+    last_updated: str
+
+
+class ApiResponse(TypedDict):
+    """CUBIC API Response Model"""
+
+    success: bool
+    results: List[ApiResult]
+
+
+def download_from_url(url: str, local_path: str) -> Optional[str]:
+    """
+    Download file from url to local_path.
+    will throw for HTTP error
+
+    :param url: CUBIC API URL string of file
+    :param local_path: local file path to save downloaded file to
+
+    :return local file path of successfully downloaded file
+    """
+    download_log = ProcessLogger(
+        "download_from_url",
+        url=url,
+        local_path=local_path,
+    )
+    download_log.log_start()
+
+    with requests.get(url, stream=True, timeout=60 * 5) as response:
+        # raise HTTP error, if occured
+        response.raise_for_status()
+        with open(local_path, "wb") as local_file:
```

Comment:
suggestion(non-blocking): not sure what the performance of this is, but [StackOverflow](https://stackoverflow.com/a/39217788) suggests this alternate implementation which is 10x faster.

Reply:
I got the following the results with a comparison of the two implementations when downloading a 25mb file: 

```
def download_from_url(url: str, local_path: str):
    with requests.get(url, stream=True, timeout=60 * 5) as response:
        # raise HTTP error, if occured
        response.raise_for_status()
        with open(local_path, "wb") as local_file:
            for file_chunk in response.iter_content(chunk_size=None):
                # chunk_size=None will write chunks in whatever size they are received
                local_file.write(file_chunk)
```
1.64 s ± 66.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

```
def alt_download_from_url(url: str, local_path: str):
    with requests.get(url, stream=True, timeout=60 * 5) as response:
        # raise HTTP error, if occured
        response.raise_for_status()
        with open(local_path, "wb") as local_file:
            shutil.copyfileobj(response.raw, local_file)
```
1.61 s ± 95 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

I think the performance improvement noted in the StackOverflow posts comes from using a small `chunk_size` in the `iter_content` implementation. With `chunk_size=None` I think the behavior ends up being very similar to the raw shutil  implementation.